src/util.rs

```diff
@@ -514,4 +564,41 @@ mod tests {
         let mut slice = b"abc".as_slice();
         assert_eq!(slice.read_cstr(), None);
     }
+
+    /// Test that we correctly binary search for a lower bound.
+    #[test]
+    fn search_lower_bound() {
```

Comment:
do you think we could incorporate exhaustive test I proposed in https://github.com/libbpf/blazesym/pull/23#issuecomment-1340100040 as another unit test, as a follow up? I'd sleep better if we had something like this :)

Reply:
Yeah, I considered doing it, but I felt like I already spent too much time on this rework. That, combined with the fact that the new set of tests is a strict improvement over basically no tests, allowed me to drop the ball on it with a clean conscience ;-) But I agree it would be beneficial.